### PR TITLE
[Wallet] [Redesign] New search text input

### PR DESCRIPTION
**Pull request by i1skn**
### Description

Implements new search text input https://www.figma.com/file/zt7aTQ5wuXycIwxq5oAmF9/Wallet--Refresh?node-id=986%3A4151

### Other changes

* Fix flickering of hamburger icon to 1px top/bottom when scrolling
* Remove Invite screen from drawer menu
* Fix incorrect error message on PinCode Screen

### Tested

Android and iPhone

### Related issues

- Related to https://github.com/celo-org/celo-monorepo/issues/3645

### Backwards compatibility

Yes

### Screenshots

Simulator Screen Shot - iPhone 11 Pro - 2020-06-23 at 13 33 47
ZxGqWZ7480WBcKtd9dxlth85SERbSdD0

### CODE REVIEW

File: packages/mobile/src/account/__snapshots__/Account.test.tsx.snap [hunk 1]
```diff
@@ -14,8 +14,10 @@ exports[`Account renders correctly 1`] = `
       Object {
         "alignItems": "center",
         "backgroundColor": "transparent",
-        "borderBottomColor": "#EDEEEF",
-        "borderBottomWidth": 0,
+        "borderBottomColor": AnimatedValue {
+          " __value": NaN,
```
Comment (i1skn) [inline packages/mobile/src/account/__snapshots__/Account.test.tsx.snap:18]: I was not able to have `0` here, it's only a problem in snapshots. Lmk if anyone know how to fix `interpolateColors` with jest.

File: packages/mobile/src/navigator/DrawerNavigator.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ import deviceInfoModule from 'react-native-device-info'
 import Account from 'src/account/Account'
 import FiatExchange from 'src/account/FiatExchange'
 import GoldEducation from 'src/account/GoldEducation'
-import { default as InviteScreen } from 'src/account/Invite'
+// import { default as InviteScreen } from 'src/account/Invite'
```
Comment (aaronmgdr) [inline packages/mobile/src/navigator/DrawerNavigator.tsx]: remove?

File: packages/mobile/src/navigator/DrawerNavigator.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ import { AddWithdraw } from 'src/icons/navigator/AddWithdraw'
 import { Gold } from 'src/icons/navigator/Gold'
 import { Help } from 'src/icons/navigator/Help'
 import { Home } from 'src/icons/navigator/Home'
-import { Invite } from 'src/icons/navigator/Invite'
+// import { Invite } from 'src/icons/navigator/Invite'
```
Comment (aaronmgdr) [inline packages/mobile/src/navigator/DrawerNavigator.tsx]: remove?

**Review by aaronmgdr — APPROVED** [review]

File: packages/mobile/src/home/SendOrRequestBar.tsx [hunk 4]
```diff
@@ -3,27 +3,27 @@ import Touchable from '@celo/react-components/components/Touchable'
...
 import { StyleSheet, View } from 'react-native'
 import { Namespaces } from 'src/i18n'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 
 export default function SendOrRequestBar() {
-  const { t } = useTranslation(Namespaces.sendFlow7)
-
-  const onPressSend = useCallback(() => {
+  function onPressSend() {
```
Comment (jmrossy) [inline packages/mobile/src/home/SendOrRequestBar.tsx]: We should be using arrow functions for inline functions right?

File: packages/mobile/src/send/SendSearchInput.tsx [hunk 5]
```diff
@@ -1,64 +1,35 @@
...
-)
 
 interface SendSearchInputProps {
-  isPhoneEnabled: boolean
   onChangeText: (value: string) => void
 }
 
 // Input field for Send screen
 export function SendSearchInput(props: SendSearchInputProps) {
+  function handleChangeText(value: string) {
```
Comment (jmrossy) [inline packages/mobile/src/send/SendSearchInput.tsx]: ditto

File: packages/mobile/src/send/SendSearchInput.tsx [hunk 6]
```diff
@@ -1,64 +1,35 @@
...
-        placeholder={isPhoneEnabled ? t('nameOrPhoneNumber') : t('walletAddress')}
-        value={input}
-        onChangeText={handleChangeText}
-        icon={isPhoneEnabled ? <Search /> : undefined}
-        title={isPhoneEnabled ? undefined : t('global:to')}
-        style={styles.textInput}
-        shouldShowClipboard={isValidAddress}
-        testID={'RecipientSearchInput'}
-      />
+      <SearchInput placeholder={t('global:search')} value={input} onChangeText={handleChangeText} />
```
Comment (jmrossy) [inline packages/mobile/src/send/SendSearchInput.tsx:24]: Does this new input still show the paste button when there's an address in the clipboard?

File: packages/mobile/src/home/SendOrRequestBar.tsx (hunk 4 above)
Comment (aaronmgdr) [reply packages/mobile/src/home/SendOrRequestBar.tsx]: if theres no "this" its seems less important. idk new debate?

Comment (i1skn) [reply packages/mobile/src/home/SendOrRequestBar.tsx]: Well, I see I was the only one who voted for using `function` for this, so, it seems yes - will refactor it.

File: packages/mobile/src/send/SendSearchInput.tsx (hunk 6 above)
Comment (i1skn) [reply packages/mobile/src/send/SendSearchInput.tsx:24]: Well, I do not see it in the design, @coreycelo can you confirm that search field at `Send` screen does not have paste functionality?

File: packages/mobile/src/navigator/DrawerNavigator.tsx (hunk 2 above)
Comment (i1skn) [reply packages/mobile/src/navigator/DrawerNavigator.tsx]: Well, I did not want to remove these references, cause I've left `Invite` screen commented down below until we can redesign it as well. So, I would either remove them all or left all the references commented. WDYT?

File: packages/mobile/src/send/SendSearchInput.tsx (hunk 6 above)
Comment (coreycelo) [reply packages/mobile/src/send/SendSearchInput.tsx:24]: Happy to include it, it's a nice Easter egg. As long as we allow the normal text paste (tap and hold to reveal the option), I don't see this as a blocker. I'll mock it up.

Comment (i1skn) [reply packages/mobile/src/send/SendSearchInput.tsx:24]: @coreycelo  Is it ok we ship it now without this functionality?

File: packages/mobile/src/navigator/DrawerNavigator.tsx (hunk 2 above)
Comment (i1skn) [reply packages/mobile/src/navigator/DrawerNavigator.tsx]: I've decided to remove it for now completely

File: packages/mobile/src/send/SendSearchInput.tsx (hunk 6 above)
Comment (coreycelo) [reply packages/mobile/src/send/SendSearchInput.tsx:24]: Sounds good to me

**Comment by i1skn** [comment]
@jmrossy PTAL, I've addressed the feedback!

**Review by jmrossy — APPROVED** [review]

**Pull request merged**